### PR TITLE
2022 theme stylings for product and shop pages.

### LINKED ISSUE #31549: Twenty Twenty Two (2022) theme compatibility for shop related pages
### Describe the solution you'd like

This issue tracks adding WooCommerce compatibility for following pages:

- [ ] Shop page
- [ ] Home page with products
- [ ] Product page - Simple
- [ ] Product page - Variable
- [ ] Product page - Downloadable
- [ ] Product page - Grouped

### Describe alternatives you've considered

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by vedanshujain**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR contains styling support for 2022 theme support for product and shop page.

Closes #31549.

### How to test the changes in this Pull Request:

1. Switch to WP 5.9 beta.
2. Switch to 2022 theme.
3. View shop and individual product page.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enhancement -  Add support for 2022 theme shop and product pages.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-two.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		return apply_filters( 'woocommerce_twenty_twenty_two_styles', $styles );
+	}
+
+	/**
+	 * Enqueue the wp-admin CSS overrides for this theme.
+	 */
+	public static function enqueue_admin_styles() {
+		wp_enqueue_style(
+			'woocommerce-twenty-twenty-one-admin',
+			str_replace( array( 'http:', 'https:' ), '', WC()->plugin_url() ) . '/assets/css/twenty-twenty-two-admin.css',
```
Comment (jeffstieler) [inline plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-two.php]: Should we create a stub source SCSS file for this so it exists?

**Review by jeffstieler — CHANGES_REQUESTED** [review]
Most of this is looking good. I noticed a few differences compared to the Figma designs:

Screen Shot 2022-01-04 at 10 08 10 AM

* Content width should be 1000px
* The breadcrumbs should not be rendered
* The horizontal divider should not be rendered (looks like this is in the notices? perhaps attach to the notices and not the wrapper)

Screen Shot 2022-01-04 at 10 18 17 AM

I think the font size for the "Shop" page title is a little larger than expected (though Figma shows that as an H3 which is incorrect..)

On the single product page:

Screen Shot 2022-01-04 at 10 39 50 AM

* Max width should be 1000px
* Zoom icon in product image is different from designs
* Font sizes do not match designs

**Comment by jeffstieler** [comment]
I've got a fix for the notices border in https://github.com/woocommerce/woocommerce/commit/43619bbf90977dd600a5c08405ef71a70ef8c55e.

**Comment by vedanshujain** [comment]
Thanks for the detailed review, much appreciated @jeffstieler 🙌 

> Content width should be 1000px

This is currently set by the theme to be 650px. Should we be overwriting this?

> The breadcrumbs should not be rendered

Addressed in https://github.com/woocommerce/woocommerce/pull/31536/commits/49b5a049efcc39f27da951e481951d66a61f2531.

> The horizontal divider should not be rendered (looks like this is in the notices? perhaps attach to the notices and not the wrapper)

Addressed in https://github.com/woocommerce/woocommerce/pull/31536/commits/49b5a049efcc39f27da951e481951d66a61f2531. Your fix works fine as well.

> I think the font size for the "Shop" page title is a little larger than expected 

Fixed in https://github.com/woocommerce/woocommerce/pull/31536/commits/6cd32f755bf0130f828137fedd13498fe0021bd7

> Zoom icon in product image is different from designs

I am bit struggling with this get this icon to show exactly as in the design. I would say current one is good enough, unless there is a neat way to do it which I am missing.

> Font sizes do not match designs

I reduced font sizes in tabs, but otherwise its not looking good in wide screens. Maybe we should have this verified again.

**Comment by jeffstieler** [comment]
> This is currently set by the theme to be 650px. Should we be overwriting this?

Oh hm, all of the designs have 1000px as the width. I see that the `wideSize` is set to `1000px`...  we should ask the designer about this.

> I am bit struggling with this get this icon to show exactly as in the design. I would say current one is good enough, unless there is a neat way to do it which I am missing.

That works for me - I ended up having the same trouble with the icon in the select2 dropdowns.

**Comment by vedanshujain** [comment]
> all of the designs have 1000px as the width. I see that the wideSize is set to 1000px

Ah you are right, looks like we need to add `.alignwide` class for the theme to set to 1000px. I will take care of this.

File: plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-two.php (hunk 1 above)
Comment (vedanshujain) [reply plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-two.php]: I guess it's not needed? I will remove it.

**Comment by vedanshujain** [comment]
@jeffstieler Added another PR https://github.com/woocommerce/woocommerce/pull/31611 with remaining fixes, although that CSS code isn't pretty at all.

**Comment by Konamiman** [comment]
**EDIT:** I was testing with a somewhat old version of the prodyct vendors extension, after upgrading to the lates one it looks fine.

Not sure if relevant for this PR in particular, but "related products" in the individual prodduct page doesn't look good when the product vendors extension is active:

image

**Review by jeffstieler — APPROVED** [review]

**Pull request merged**